.pre-commit-config.yaml

```diff
@@ -0,0 +1,85 @@
+default_language_version:
+  python: python3
+
+ci:
+  autofix_commit_msg: |
+    [pre-commit.ci] auto fixes from pre-commit.com hooks
+  autofix_prs: true
+  autoupdate_branch: "master"
+  autoupdate_commit_msg: "[pre-commit.ci] pre-commit autoupdate"
+  autoupdate_schedule: quarterly
+  skip: []
+  submodules: false
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: check-yaml
+      - id: check-json
+      - id: check-executables-have-shebangs
+      - id: check-toml
+      - id: check-docstring-first
+      - id: check-added-large-files
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+      - id: check-case-conflict
+      - id: mixed-line-ending
+      - id: end-of-file-fixer
+      - id: check-case-conflict
+      - id: forbid-new-submodules
+      - id: pretty-format-json
+        args: ["--autofix", "--no-sort-keys", "--indent=4"]
+
+
+
+  -   repo: https://github.com/asottile/blacken-docs
+      rev: 1.15.0
+      hooks:
+      -   id: blacken-docs
+          additional_dependencies: [black==23.3.0]
+          exclude: ^.github/
+
+  - repo: https://github.com/PyCQA/isort
```

Comment:
I think it could be better to use only ruff and remove isort and flake8 precommit, since ruff could do all the job. I could help you with that.

Reply:
Okay.. Changing everything in favor of ruff.